python/tests/test_schema.py

```diff
@@ -415,3 +415,133 @@ def test_deserialize_schema(table_schema_simple: Schema):
+    assert prune_columns(table_schema_nested, {5}, True) == Schema(
+        NestedField(
+            field_id=4,
+            name="qux",
+            field_type=ListType(type="list", element_id=5, element_type=StringType(), element_required=True),
+            required=True,
+        ),
+        schema_id=1,
+        identifier_field_ids=[1],
+    )
+
+
+def test_prune_columns_map(table_schema_nested: Schema):
+    assert prune_columns(table_schema_nested, {9}, False) == Schema(
+        NestedField(
+            field_id=6,
+            name="quux",
+            field_type=MapType(
+                type="map",
+                key_id=7,
+                key_type=StringType(),
+                value_id=8,
+                value_type=MapType(
+                    type="map", key_id=9, key_type=StringType(), value_id=10, value_type=IntegerType(), value_required=True
+                ),
+                value_required=True,
+            ),
+            required=True,
+        ),
+        schema_id=1,
+        identifier_field_ids=[1],
+    )
+
+
+def test_prune_columns_map_full(table_schema_nested: Schema):
+    assert prune_columns(table_schema_nested, {9}, True) == Schema(
+        NestedField(
+            field_id=6,
+            name="quux",
+            field_type=MapType(
+                type="map",
+                key_id=7,
+                key_type=StringType(),
+                value_id=8,
+                value_type=MapType(
+                    type="map", key_id=9, key_type=StringType(), value_id=10, value_type=IntegerType(), value_required=True
+                ),
+                value_required=True,
+            ),
+            required=True,
+        ),
+        schema_id=1,
+        identifier_field_ids=[1],
+    )
+
+
+def test_prune_columns_map_key(table_schema_nested: Schema):
+    assert prune_columns(table_schema_nested, {10}, False) == Schema(
+        NestedField(
+            field_id=6,
+            name="quux",
+            field_type=MapType(
+                type="map",
+                key_id=7,
+                key_type=StringType(),
+                value_id=8,
+                value_type=MapType(
+                    type="map", key_id=9, key_type=StringType(), value_id=10, value_type=IntegerType(), value_required=True
+                ),
+                value_required=True,
+            ),
+            required=True,
+        ),
+        schema_id=1,
+        identifier_field_ids=[1],
+    )
+
+
+def test_prune_columns_struct(table_schema_nested: Schema):
+    assert prune_columns(table_schema_nested, {16}, False) == Schema(
+        NestedField(
+            field_id=15,
+            name="person",
+            field_type=StructType(NestedField(field_id=16, name="name", field_type=StringType(), required=False)),
+            required=False,
+        ),
+        schema_id=1,
+        identifier_field_ids=[1],
+    )
+
+
+def test_prune_columns_struct_full(table_schema_nested: Schema):
```

Comment:
I think you also need a test for this case with `select_full_types=False` where the field type is an empty struct.

Reply:
Done 👍🏻

Reply:
@Fokko, I don't see this anywhere. Are you sure you pushed a version with it?

Reply:
They are here: https://github.com/apache/iceberg/blob/master/python/tests/test_schema.py#L565-L590 Let me know if I misinterpreted it.